### ISSUE TITLE
[BUG]: Not able to run zenml pipeline from Jenkins PodTemplates to kubernetes target cluster defined into zenml stack

### ISSUE THREAD

**Issue by leefromscratch** [zenml-io/zenml#1396]
### System Information

ZENML_LOCAL_VERSION: 0.35.0
ZENML_SERVER_VERSION: 0.35.0
ZENML_SERVER_DATABASE: mysql
ZENML_SERVER_DEPLOYMENT_TYPE: aws
ZENML_CONFIG_DIR: /root/.config/zenml
ZENML_LOCAL_STORE_DIR: /root/.config/zenml/local_stores
ZENML_SERVER_URL: https://xxxxxxxxxxxxxxxxxxxxxxxx
ZENML_ACTIVE_REPOSITORY_ROOT: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
PYTHON_VERSION: 3.8.10
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'linux', 'linux_distro': 'ubuntu', 'linux_distro_like': 'debian', 'linux_distro_version': '20.04'}
ACTIVE_WORKSPACE: default
ACTIVE_STACK: zenml-report
ACTIVE_USER: default
TELEMETRY_STATUS: enabled
ANALYTICS_CLIENT_ID: xxxxxxxxxxxxxxxxxxxxxx
ANALYTICS_USER_ID: xxxxxxxxxxxxxxxxxxxxxxxx
ANALYTICS_SERVER_ID: xxxxxxxxxxxxxxxxxxxxxxxxxx
INTEGRATIONS: ['aws', 'kaniko', 'kubernetes', 'mlflow', 'pillow', 's3', 'scipy', 'sklearn']
PACKAGES: {'brotli': '1.0.9', 'flask': '2.2.3', 'gitpython': '3.1.31', 'jinja2': '3.1.2', 'mako': '1.2.4', 'markdown': '3.4.1', 'markupsafe': '2.1.2', 'pillow': '9.4.0', 'pyjwt': '2.6.0', 'pymysql':
'1.0.2', 'pyyaml': '5.4.1', 'pygments': '2.14.0', 'sqlalchemy': '1.4.41', 'sqlalchemy-utils': '0.38.3', 'send2trash': '1.8.0', 'unidecode': '1.3.6', 'werkzeug': '2.2.3', 'aiobotocore': '2.4.2', 'aiofiles':
'23.1.0', 'aiohttp': '3.8.4', 'aioitertools': '0.11.0', 'aiokafka': '0.8.0', 'aiosignal': '1.3.1', 'alembic': '1.8.1', 'analytics-python': '1.4.0', 'anyio': '3.6.2', 'argon2-cffi': '21.3.0',
'argon2-cffi-bindings': '21.2.0', 'arrow': '1.2.3', 'asttokens': '2.2.1', 'async-timeout': '4.0.2', 'attrs': '22.2.0', 'backcall': '0.2.0', 'backoff': '1.10.0', 'bcrypt': '4.0.1', 'beautifulsoup4':
'4.11.2', 'bleach': '6.0.0', 'boto3': '1.24.59', 'botocore': '1.27.59', 'cachetools': '5.3.0', 'certifi': '2022.12.7', 'cffi': '1.15.1', 'charset-normalizer': '3.0.1', 'click': '8.1.3', 'click-params':
'0.3.0', 'cloudpickle': '2.2.1', 'comm': '0.1.2', 'commonmark': '0.9.1', 'contextlib2': '21.6.0', 'contourpy': '1.0.7', 'cycler': '0.11.0', 'databricks-cli': '0.17.4', 'debugpy': '1.6.6', 'decorator':
'5.1.1', 'defusedxml': '0.7.1', 'dill': '0.3.6', 'distro': '1.8.0', 'docker': '6.0.1', 'entrypoints': '0.4', 'et-xmlfile': '1.1.0', 'executing': '1.2.0', 'fastapi': '0.89.1', 'fastapi-utils': '0.2.1',
'fastjsonschema': '2.16.2', 'fonttools': '4.38.0', 'fqdn': '1.5.1', 'frozenlist': '1.3.3', 'fsspec': '2022.11.0', 'gevent': '22.10.2', 'geventhttpclient': '2.0.2', 'gitdb': '4.0.10', 'google-auth':
'2.16.1', 'google-pasta': '0.2.0', 'greenlet': '2.0.2', 'grpcio': '1.51.3', 'gunicorn': '20.1.0', 'h11': '0.14.0', 'httplib2': '0.19.1', 'idna': '3.4', 'importlib-metadata': '4.13.0', 'importlib-resources':
'5.12.0', 'ipykernel': '6.21.2', 'ipython': '8.10.0', 'ipython-genutils': '0.2.0', 'ipywidgets': '7.7.3', 'isoduration': '20.11.0', 'itsdangerous': '2.1.2', 'jedi': '0.18.2', 'jmespath': '1.0.1', 'joblib':
'1.2.0', 'jsonpointer': '2.3', 'jsonschema': '4.17.3', 'jupyter-client': '8.0.3', 'jupyter-core': '5.2.0', 'jupyter-events': '0.6.3', 'jupyter-server': '2.3.0', 'jupyter-server-terminals': '0.4.4',
'jupyterlab-pygments': '0.2.2', 'jupyterlab-widgets': '1.1.2', 'kafka-python': '2.0.2', 'kiwisolver': '1.4.4', 'kubernetes': '18.20.0', 'llvmlite': '0.39.1', 'lxml': '4.9.2', 'matplotlib': '3.7.0',
'matplotlib-inline': '0.1.6', 'mistune': '2.0.5', 'mlflow': '2.1.1', 'mlserver': '1.2.3', 'mlserver-mlflow': '1.2.3', 'monotonic': '1.6', 'multidict': '6.0.4', 'multiprocess': '0.70.14', 'nbclassic':
'0.5.2', 'nbclient': '0.7.2', 'nbconvert': '7.2.9', 'nbformat': '5.7.3', 'nest-asyncio': '1.5.6', 'notebook': '6.5.2', 'notebook-shim': '0.2.2', 'numba': '0.56.4', 'numpy': '1.23.5', 'oauthlib': '3.2.2',
'openpyxl': '3.1.1', 'orjson': '3.8.6', 'packaging': '22.0', 'pandas': '1.5.3', 'pandocfilters': '1.5.0', 'parso': '0.8.3', 'passlib': '1.7.4', 'pathos': '0.3.0', 'pexpect': '4.8.0', 'pickleshare': '0.7.5',
'pip': '22.3.1', 'pkgutil-resolve-name': '1.3.10', 'platformdirs': '3.0.0', 'pox': '0.3.2', 'ppft': '1.7.6.6', 'prometheus-client': '0.16.0', 'prompt-toolkit': '3.0.37', 'protobuf': '3.20.3',
'protobuf3-to-dict': '0.1.5', 'psutil': '5.9.4', 'psycopg2-binary': '2.9.5', 'ptyprocess': '0.7.0', 'pure-eval': '0.2.2', 'py-grpc-prometheus': '0.7.0', 'pyarrow': '10.0.1', 'pyasn1': '0.4.8',
'pyasn1-modules': '0.2.8', 'pycparser': '2.21', 'pydantic': '1.10.5', 'pyparsing': '2.4.7', 'pyrsistent': '0.19.3', 'python-dateutil': '2.8.2', 'python-dotenv': '1.0.0', 'python-json-logger': '2.0.7',
'python-rapidjson': '1.9', 'python-terraform': '0.10.1', 'pytz': '2022.7.1', 'pyzmq': '25.0.0', 'querystring-parser': '1.2.4', 'regex': '2022.10.31', 'requests': '2.28.2', 'requests-oauthlib': '1.3.1',
'rfc3339-validator': '0.1.4', 'rfc3986-validator': '0.1.1', 'rich': '12.6.0', 'rsa': '4.9', 's3fs': '2022.11.0', 's3transfer': '0.6.0', 'sagemaker': '2.117.0', 'schema': '0.7.5', 'scikit-learn': '1.2.1',
'scipy': '1.10.1', 'setuptools': '66.1.1', 'shap': '0.41.0', 'six': '1.16.0', 'slicer': '0.0.7', 'smdebug-rulesconfig': '1.0.1', 'smmap': '5.0.0', 'sniffio': '1.3.0', 'soupsieve': '2.4',
'sqlalchemy2-stubs': '0.0.2a32', 'sqlmodel': '0.0.8', 'sqlparse': '0.4.3', 'stack-data': '0.6.2', 'starlette': '0.22.0', 'starlette-exporter': '0.15.1', 'tabulate': '0.9.0', 'terminado': '0.17.1',
'threadpoolctl': '3.1.0', 'tinycss2': '1.2.1', 'tornado': '6.2', 'tqdm': '4.64.1', 'traitlets': '5.9.0', 'tritonclient': '2.30.0', 'typing-extensions': '4.5.0', 'uri-template': '1.2.0', 'urllib3':
'1.26.14', 'uvicorn': '0.20.0', 'uvloop': '0.17.0', 'validators': '0.18.2', 'wcwidth': '0.2.6', 'webcolors': '1.12', 'webencodings': '0.5.1', 'websocket-client': '1.5.1', 'wheel': '0.38.4',
'widgetsnbextension': '3.6.2', 'wrapt': '1.14.1', 'yarl': '1.8.2', 'zenml': '0.35.0', 'zipp': '3.14.0', 'zope.event': '4.6', 'zope.interface': '5.5.2'}

### What happened?

we are attempting to run our pipeline.py from Jenkins which is configured to use podTemplate ( kubernetes plugin ).
Our Zenml server is deployed into another EKS cluster. 
Those are different steps we are following : 
- Set kubeconfig
- zenml connect
- zenml set stack 
- run pipeline.py

We checked that we are able to execute some kubectl commands from this "Podtemplate" to target kubernetes cluster defined into our zenml stack. 

But instead of using configuration from zenml stack, pipeline.py is trying to be executed on the hosted kubernetes cluster of our Jenkins. 

Our kubeconfig :
 
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: xxxx
  name: kubernetes_context: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
contexts:
- context:
    cluster: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
    user: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
  name: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
current-context: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
kind: Config
preferences: {}
users:
- name: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - xxxxxxxx
      - eks
      - get-token
      - --cluster-name
      - zenml-dev-eks
      command: aws

components:
    ....
    orchestrator:
        configuration:
            kubernetes_context: arn:aws:eks:xxxxxxxx:xxxxxxxxxx:cluster/zenml-dev-eks
        flavor: kubernetes
        name: xxxxxxxxx
        type: orchestrator

### Reproduction steps

Those steps should be run from Jenkins PodTemplate. 
Container should contains zenml cli, aws, python
1- aws eks update-kubeconfig --name zenml-dev-eks
2. zenml connect 
3. zenml set stack
4. run pipeline.py

### Relevant log output

```shell
Zenml pod is running on this hosted jenkins kubernetes.
```

### Expected

Pipeline should be run to target kubernetes defined into stack definition instead of current EKS which hosts our Jenkins. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by stefannica** [comment]
This is definitely linked to a problem in the ZenML kubernetes orchestrator. The client is trying to use the in-cluster k8s configuration over the one configured in the stack component:

```
# from kubernetes_orchestrator.py:

    def _initialize_k8s_clients(self) -> None:
        """Initialize the Kubernetes clients."""
        kube_utils.load_kube_config(context=self.config.kubernetes_context)
        self._k8s_core_api = k8s_client.CoreV1Api()
        self._k8s_batch_api = k8s_client.BatchV1beta1Api()
        self._k8s_rbac_api = k8s_client.RbacAuthorizationV1Api()

# from kube_utils.py:

def load_kube_config(context: Optional[str] = None) -> None:
    """Load the Kubernetes client config.

    Depending on the environment (whether it is inside the running Kubernetes
    cluster or remote host), different location will be searched for the config
    file.

    Args:
        context: Name of the Kubernetes context. If not provided, uses the
            currently active context.
    """
    try:
        k8s_config.load_incluster_config()
    except k8s_config.ConfigException:
        k8s_config.load_kube_config(context=context)

```
